crates/ruff/src/noqa.rs

```diff
@@ -248,22 +248,41 @@ impl FileExemption {
                     let path_display = relativize_path(path);
                     warn!("Invalid `# ruff: noqa` directive at {path_display}:{line}: {err}");
                 }
-                Ok(Some(ParsedFileExemption::All)) => {
-                    return Some(Self::All);
-                }
-                Ok(Some(ParsedFileExemption::Codes(codes))) => {
-                    exempt_codes.extend(codes.into_iter().filter_map(|code| {
-                        if let Ok(rule) = Rule::from_code(get_redirect_target(code).unwrap_or(code))
-                        {
-                            Some(rule.noqa_code())
-                        } else {
-                            #[allow(deprecated)]
-                            let line = locator.compute_line_index(range.start());
-                            let path_display = relativize_path(path);
-                            warn!("Invalid code provided to `# ruff: noqa` at {path_display}:{line}: {code}");
-                            None
+                Ok(Some(exemption)) => {
+                    if !locator
+                        .slice(TextRange::new(
```

Comment:
Could we use indentation_at_offset here? It is okay if it is Some, None means the directive is not an own line comment. Or would that be confusing

Nit: it's a bit unfortunate that line_start first does a backwards scan finding the start of the line, just to do a forward scan right after